tiledb/sm/storage_manager/context.cc

```diff
@@ -127,12 +100,46 @@ stats::Stats* Context::stats() const {
   return stats_.get();
 }
 
-Status Context::init_thread_pools(Config* const config) {
-  // If `config` is null, use a default-constructed config.
-  Config tmp_config;
-  if (config != nullptr)
-    tmp_config = *config;
+Status Context::init(const Config& config) {
+  RETURN_NOT_OK(init_loggers(config));
+
+  // Register stats.
+  stats::all_stats.register_stats(stats_);
+
+  // Create storage manager
+  storage_manager_ = new (std::nothrow)
+      tiledb::sm::StorageManager(&compute_tp_, &io_tp_, stats_.get(), logger_);
+  if (storage_manager_ == nullptr)
+    return logger_->status(Status_ContextError(
+        "Cannot initialize context Storage manager allocation failed"));
+
+  // Initialize storage manager
+  auto sm = storage_manager_->init(config);
```

Comment:
This is the only place where `StorageManager::init()` is ever called. It would be very simple just to inline that function in `StorageManager::StorageManager` and make it a C.41 class as well.

If it were more complicated, I'd say it might be out of scope for this PR, but it's a quite small change.

Reply:
I started to do this but ran into some issues that made it not quite small (I don't remember what the issues were, but there were some).  I opened a story about this.